testcases/OpTestDropbearSafety.py

```diff
@@ -89,38 +89,13 @@ def __init__(self, i_bmcIP, i_bmcUser, i_bmcPasswd,
     def test_dropbear_running(self):
         print "Test Dropbear running in Petitboot"
         print "Performing IPMI Power Off Operation"
-        try:
-            # Perform a IPMI Power OFF Operation(Immediate Shutdown)
-            self.cv_IPMI.ipmi_power_off()
-            if int(self.cv_SYSTEM.sys_wait_for_standby_state(BMC_CONST.SYSTEM_STANDBY_STATE_DELAY)) == 0:
-                print "System is in standby/Soft-off state"
-            else:
-                l_msg = "System failed to reach standby/Soft-off state"
-                raise OpTestError(l_msg)
-            self.cv_IPMI.ipmi_power_on()
-            self.console = self.cv_SYSTEM.sys_get_ipmi_console()
-
-            # Exiting to petitboot shell
-            self.console.expect('Petitboot', timeout=BMC_CONST.PETITBOOT_TIMEOUT)
-            # Exiting to petitboot
-            self.console.sendcontrol('l')
-            self.console.send('\x1b[B')
-            self.console.send('\x1b[B')
-            self.console.send('\r')
-            self.console.expect('Exiting petitboot')
-            self.console.send('\r')
-            self.console.send('\x08')
-            self.cv_IPMI.ipmi_host_set_unique_prompt(self.console)
-            self.cv_IPMI.run_host_cmd_on_ipmi_console("uname -a")
-            res = self.cv_IPMI.run_host_cmd_on_ipmi_console("ps")
-            if res[-1] == '0':
-                print 'ps command worked'
-            else:
-                raise OpTestError('failed to run ps command')
-
-        except OpTestError:
-            self.cv_SYSTEM.sys_bmc_power_on_validate_host()
-        self.cv_SYSTEM.sys_bmc_power_on_validate_host()
```

Comment:
I am facing below problem on a habanero system which is mentioned in this issue https://github.com/open-power/op-build/issues/672


15:18:48 # Starting Test Inband IPMI USB Interface Functionality.5
17:32:36 # 0
17:32:36 # Starting Test Out-of-band IPMI.6
17:53:32 # 0
17:53:32 # Starting Test IPMI System Power Control.7
18:07:34 # 18:07:34 :: ERROR: command failed with exit code 256
18:07:34 # 256
18:07:34 
18:07:34 #   Failed test 'Test IPMI System Power Control.7'
18:07:34 #   at ./run-op-it line 439.
18:07:34 # Starting Test IPMI Lock/Unlock Functionality.8
18:15:09 # 18:15:09 :: ERROR: command failed with exit code 256
18:15:09 # 256
18:15:09 
18:15:09 #   Failed test 'Test IPMI Lock/Unlock Functionality.8'
18:15:09 #   at ./run-op-it line 439.
18:15:09 # Starting Test Timer Facility Error Injections.9
18:22:45 # 18:22:45 :: ERROR: command failed with exit code 256
18:22:45 # 256
18:22:45 
18:22:45 #   Failed test 'Test Timer Facility Error Injections.9'
18:22:45 #   at ./run-op-it line 439.
18:22:45 # Starting Test Chip TOD Error Injections.10
18:30:20 # 18:30:20 :: ERROR: command failed with exit code 256
18:30:20 # 256
18:30:20 
18:30:20 #   Failed test 'Test Chip TOD Error Injections.10'
18:30:20 #   at ./run-op-it line 439.
18:30:20 # Starting Test PRD(Processor Runtime Diagnostics) Driver.11
18:37:56 # 18:37:56 :: ERROR: command failed with exit code 256
18:37:56 # 256
18:37:56 
18:37:56 #   Failed test 'Test PRD(Processor Runtime Diagnostics) Driver.11'
18:37:56 #   at ./run-op-it line 439.
18:37:56 # Starting Test HMI Processor Recovery Done.12
18:45:31 # 18:45:31 :: ERROR: command failed with exit code 256
18:45:31 # 256
18:45:31 
18:45:31 #   Failed test 'Test HMI Processor Recovery Done.12'
18:45:31 #   at ./run-op-it line 439.
18:45:31 # Starting Test HMI Processor Recovery Error Masked.13
18:53:07 # 18:53:07 :: ERROR: command failed with exit code 256
18:53:07 # 256
18:53:07 
18:53:07 #   Failed test 'Test HMI Processor Recovery Error Masked.13'
18:53:07 #   at ./run-op-it line 439.
18:53:07 # Starting Test HMI Malfunction Alert.14
19:00:42 # 19:00:42 :: ERROR: command failed with exit code 256
19:00:42 # 256
19:00:42 
19:00:42 #   Failed test 'Test HMI Malfunction Alert.14'
19:00:42 #   at ./run-op-it line 439.
19:00:42 # Starting Test Clear Gard Entries.15
19:12:24 # 19:12:24 :: ERROR: command failed with exit code 256
19:12:24 # 256
19:12:24 
19:12:24 #   Failed test 'Test Clear Gard Entries.15'
19:12:24 #   at ./run-op-it line 439.
19:12:24 # Starting Test HMI Hypervisor Resource Error.16
19:19:59 # 19:19:59 :: ERROR: command failed with exit code 256
19:19:59 # 256
19:19:59 
19:19:59 #   Failed test 'Test HMI Hypervisor Resource Error.16'
19:19:59 #   at ./run-op-it line 439.
19:19:59 # Starting Test Clear Gard Entries.17
19:31:41 # 19:31:40 :: ERROR: command failed with exit code 256
19:31:41 # 256
19:31:41 
19:31:41 #   Failed test 'Test Clear Gard Entries.17'
19:31:41 #   at ./run-op-it line 439.
19:31:41 # Starting Test I2C Driver Functionality.18
19:39:16 # 19:39:16 :: ERROR: command failed with exit code 256
19:39:16 # 256
19:39:16 
19:39:16 #   Failed test 'Test I2C Driver Functionality.18'
19:39:16 #   at ./run-op-it line 439.
19:39:16 # Starting Test Atmel(at24) Driver Functionality.19
19:46:52 # 19:46:51 :: ERROR: command failed with exit code 256
19:46:52 # 256
19:46:52 
19:46:52 #   Failed test 'Test Atmel(at24) Driver Functionality.19'
19:46:52 #   at ./run-op-it line 439.
19:46:52 # Starting Test RTC Driver Functionality.20
19:54:27 # 19:54:27 :: ERROR: command failed with exit code 256
19:54:27 # 256
19:54:27 
19:54:27 #   Failed test 'Test RTC Driver Functionality.20'
19:54:27 #   at ./run-op-it line 439.
19:54:27 # Starting Test Hwmon Driver Functionality.21
20:02:02 # 20:02:02 :: ERROR: command failed with exit code 256
20:02:02 # 256
20:02:02 
20:02:02 #   Failed test 'Test Hwmon Driver Functionality.21'
20:02:02 #   at ./run-op-it line 439.
20:02:02 # Starting Test Swicth Endian Syscall.22
20:09:38 # 20:09:38 :: ERROR: command failed with exit code 256
20:09:38 # 256
20:09:38 
20:09:38 #   Failed test 'Test Swicth Endian Syscall.22'
20:09:38 #   at ./run-op-it line 439.
20:09:38 # Starting Test IPMI Heartbeat Functionality.23
20:17:13 # 20:17:13 :: ERROR: command failed with exit code 256
20:17:13 # 256
20:17:13 
20:17:13 #   Failed test 'Test IPMI Heartbeat Functionality.23'
20:17:13 #   at ./run-op-it line 439.
20:17:13 # Starting Test MTD PNOR Driver Functionality.24
20:24:49 # 20:24:49 :: ERROR: command failed with exit code 256
20:24:49 # 256
20:24:49 
20:24:49 #   Failed test 'Test MTD PNOR Driver Functionality.24'
20:24:49 #   at ./run-op-it line 439.
20:24:49 # Starting Test In-band IPMI Open Interface Functionality.25
20:32:24 # 20:32:24 :: ERROR: command failed with exit code 256
20:32:24 # 256
20:32:24 
20:32:24 #   Failed test 'Test In-band IPMI Open Interface Functionality.25'
20:32:24 #   at ./run-op-it line 439.
20:32:24 # Starting Test BMC Cold Reset Boot Sequence.26
20:40:59 # 0
20:40:59 # Starting Test BMC Warm Reset Boot Sequence.27
20:48:53 # 0
20:48:53 # Starting Test Fan Control Enable Functionality.28
20:49:08 # 0
20:49:08 # Starting Test Fan Control Disable Functionality.29
20:49:23 # 0
20:49:23 # Starting Test System Power Restore Policy always_on.30
20:55:30 # 0
20:55:30 # Starting Test System Power Restore Policy always_off.31
21:04:10 # 0
21:04:10 # Starting Test System Power Restore Policy previous.32
21:14:23 # 0
21:14:23 # Starting 33.33


If you look at the complete execution system is failed to boot at exactly at test no. 7 initially Test IPMI System Power Control.7 (Here at this point normal power off/on is not bringing the system up) . Again test is passed at  Test BMC Cold Reset Boot Sequence.26 . In b/w all the tests are failed as we are not able to bring up. So better solution is we need to use BMC cold reset boot sequence.
 https://github.com/open-power/op-test-framework/pull/77